### ISSUE TITLE
Default datetime format is incorrect

### ISSUE THREAD

**Issue by kirlat** [lifeinthegrid/duplicator#21]
It seems that in SQL export file generated by Duplicated a default datetime value is '0000-00-00 00:00:00'. I believe this is wrong, because, according to MySQL specs (https://dev.mysql.com/doc/refman/5.7/en/datetime.html), such value is not allowed:

> The DATETIME type is used for values that contain both date and time parts. MySQL retrieves and displays DATETIME values in 'YYYY-MM-DD HH:MM:SS' format. The supported range is '1000-01-01 00:00:00' to '9999-12-31 23:59:59'.

I had this issue in several tables, including comments, posts and several others. This was causing a table creation error, and, as a result of this, those tables were not created and nothing was imported. I changed those values manually to be '1000-01-01 00:00:00' and it fixed the issue, but it's definitely not a good way to deal with it for an average user.

If would be great if Duplicator make sure that default values for such fields are within an allowed range. Thanks!

**Comment by lifeinthegrid** [comment]
Hey Kirlat,

 Right now the code that generates the default values is created by both a "SHOW CREATE TABLE"
https://dev.mysql.com/doc/refman/5.7/en/show-create-table.html

or if you enabled the mysqldump option then that application will handle creating the tables.   I'll have to do some research to see if either of those options control the way in which the columns can actually be changed.   

I think what may be happening is that when you build a package on Site A it has the datetime stamp setup to support TableXYZ as ''0000-00-00 00:00:00'  then when you move it to Site B its not setup to support the zero type format and then the error.

You may also try playing with the SQL mode on Step 2 of the installer under the Options section.  Possibly a mode such as NO_ZERO_DATE Will work.
https://dev.mysql.com/doc/refman/5.7/en/sql-mode.html#sqlmode_no_zero_date

We have had several people report success with just using:
NO_ENGINE_SUBSTITUTION
https://dev.mysql.com/doc/refman/5.7/en/sql-mode.html#sqlmode_no_engine_substitution

However I think the best solution would be if we could auto-correct this for users.  We do have some items in the back-log to look at this.   

If you happen to have PHP knowledge and have some spare time we definitely would love the help...   If not we will try and get to it at some point...

Thanks for the feedback!

**Comment by lifeinthegrid** [comment]
Hey Kirlat,

After some research it looks like this issue is more prevalent when using the PHP build mode vs mysqldump.  If you can give the developer version a try it may have the updates for your issue.  The very latest version (1.2.11) can be found by following these instructions:

	> [Download Latest Version](https://downloads.wordpress.org/plugin/duplicator.zip)
	> Uninstall your current version and install this one.
	
Let me know if this solves your issue~

Thanks

**Comment by kirlat** [comment]
Hi @lifeinthegrid!

Sorry for a super late response, but I was very busy with my work duties that, unfortunately, prevailed over everything else. I've tried just the latest version of Duplicator plugin (1.2.18) and it seems to have no problem I've mentioned before. I see that datetime columns still have  '0000-00-00 00:00:00' defaults, but this does not cause any error within duplicator installer this time. I checked installer log and it seems that Duplicator is using 'ONLY_FULL_GROUP_BY,STRICT_TRANS_TABLES,NO_ZERO_IN_DATE,NO_ZERO_DATE,ERROR_FOR_DIVISION_BY_ZERO,NO_AUTO_CREATE_USER,NO_ENGINE_SUBSTITUTION' SQLMODE. Is that what did the trick? I'm quite happy that it works now anyway.

Thank you guys for your great support and your effort in developing this great plugin. BTW, the new 'DB only' mode is great too!
